src/components/logical-expression-builder/input-components/text-input/text-input.component.ts

```diff
@@ -0,0 +1,32 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+@Component({
+    selector: 'ux-text-input',
+    templateUrl: './text-input.component.html'
+})
+export class TextInputComponent {
+    @Output() valueChange: EventEmitter<string> = new EventEmitter<string>();
+    @Output() valid: EventEmitter<boolean> = new EventEmitter<boolean>();
```

Comment:
Perhaps `validChange` would be a better name for the output. `valid` is a state, whereas an output is an event, so the name should indicate that. This would also allow us to add a `valid` input in future if that was ever a requirement